lib/membrane_ffmpeg_video_filter/text_overlay.ex

```diff
@@ -140,29 +141,29 @@ defmodule Membrane.FFmpeg.VideoFilter.TextOverlay do
   end
 
   defp apply_filter_if_needed(
-         %{metadata: metadata} = buffer,
+         buffer,
          ctx,
-         %{native_state: native_state, text_intervals: [{interval, _text} | streams]} = state
+         %{native_state: native_state, text_intervals: [{interval, _text} | intervals]} = state
        ) do
     cond do
-      frame_before_interval?(metadata, interval) ->
+      frame_before_interval?(buffer, interval) ->
```

Comment:
I've noticed a problem here - if buffer doesn't have pts (i.e. `pts == nil`) it will always be considered after interval (unless `to` is `:infinity`). For that case, we should only allow overlay using `text` option and/or enforce `{0, :infinity}` interval for such input, plus raise if there's a specified interval and no timestamp